actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -1304,44 +1314,74 @@ private void updateKeyCloakUserBase(Map<String, Object> userMap) {
    */
   @SuppressWarnings("unchecked")
   private void createUser(Request actorMessage) {
-    ProjectLogger.log("create user method started..");
-    Util.DbInfo usrDbInfo = Util.dbInfoMap.get(JsonKey.USER_DB);
-    Util.DbInfo addrDbInfo = Util.dbInfoMap.get(JsonKey.ADDRESS_DB);
     Map<String, Object> req = actorMessage.getRequest();
-    Map<String, Object> requestMap = null;
     Map<String, Object> userMap = (Map<String, Object>) req.get(JsonKey.USER);
-    actorMessage.getRequest().putAll(userMap);
-    UserRequestValidator.validateCreateUser(actorMessage);
-    Util.checkPhoneUniqueness(userMap, JsonKey.CREATE);
-    Util.checkEmailUniqueness(userMap, JsonKey.CREATE);
-    Map<String, Object> emailTemplateMap = new HashMap<>(userMap);
-    if (userMap.containsKey(JsonKey.WEB_PAGES)) {
-      SocialMediaType.validateSocialMedia(
-          (List<Map<String, String>>) userMap.get(JsonKey.WEB_PAGES));
-    }
-    userMap.put(JsonKey.CREATED_BY, req.get(JsonKey.REQUESTED_BY));
     // remove these fields from req
     userMap.remove(JsonKey.ENC_EMAIL);
     userMap.remove(JsonKey.ENC_PHONE);
     userMap.remove(JsonKey.EMAIL_VERIFIED);
-    // Will ignore these fields in create api
-    userMap.remove(JsonKey.EXTERNAL_ID);
-    userMap.remove(JsonKey.PROVIDER);
-    userMap.remove(JsonKey.EXTERNAL_ID_PROVIDER);
-    userMap.remove(JsonKey.ID_TYPE);
-    userMap.remove(JsonKey.EXTERNAL_ID_TYPE);
+    userMap.put(JsonKey.CREATED_BY, req.get(JsonKey.REQUESTED_BY));
+    actorMessage.getRequest().putAll(userMap);
     try {
+      String channel = (String) userMap.get(JsonKey.CHANNEL);
       // validate channel and set rootOrgId of user
-      if (StringUtils.isBlank((String) userMap.get(JsonKey.CHANNEL))) {
-        userMap.put(JsonKey.CHANNEL, ProjectUtil.getConfigValue(JsonKey.SUNBIRD_DEFAULT_CHANNEL));
+      if (StringUtils.isBlank(channel)) {
+        channel = ProjectUtil.getConfigValue(JsonKey.SUNBIRD_DEFAULT_CHANNEL);
```

Comment:
Read first from system settings service. If null, read env variable. If still null throw error in user creation.